### PR TITLE
Semihalf 2021q4 updates

### PR DESCRIPTION
**Pull request by semihalf-wojtas-marcin**

### CODE REVIEW

**Comment by semihalf-wojtas-marcin** [comment]
@emaste Hi, please review the ASLR patch.

**Comment by emaste** [comment]
ASLR description seems fine to me although I would expand on the "merged to head" bit - i.e., mention explicitly the current state of being enabled by default on 64-bit architectures [in head].

File: 2021q4/ena.adoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Contact: Dawid Gorecki <dgr@semihalf.com> +
+Contact: Marcin Wojtas <mw@semihalf.com>
+
+ENA (Elastic Network Adapter) is the smart NIC available in the virtualized environment of Amazon Web Services (AWS).
+The ENA driver supports multiple transmit and receive queues and can handle up to 100 Gb/s of network traffic, depending on the instance type on which it is used.
+
+Completed since the last update:
+
+* Add IPv6 L4 checksum offload support to the driver
+* Add NUMA awareness to the driver whenever kernel RSS option is being enabled
```
Comment (pauamma) [inline 2021q4/ena.adoc:16]: "when the RSS kernel option is enabled", I think.

**Review by pauamma — CHANGES_REQUESTED** [review]
Other than  that, LGTM.

File: 2021q4/aslr.adoc [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Contact: Marcin Wojtas <mw@semihalf.com>
+
+Address Space Layout Randomization (ASLR) is an exploit mitigation
+technique implemented in the majority of modern operating systems.
+It involves randomly positioning the base address of an executable
+and the position of libraries, heap, and stack, in a process's address
+space. Although over the years ASLR proved to not guarantee full OS
+security on its own, this mechanism can make exploitation more difficult.
+
+Semihalf team made an effort to toggle the kernel settings of all 64-bit
```
Comment (pauamma) [inline 2021q4/aslr.adoc:13]: s/Semihalf/The Semihalf/

By "toggle the kernel settings...", do you mean enable the option by default"?

**Review by pauamma — CHANGES_REQUESTED** [review]

File: 2021q4/ls1027a.adoc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+Contact: Kornel Dulęba <mindal@semihalf.com> +
+Contact: Artur Rojek <ar@semihalf.com> +
+Contact: Hubert Mazur <hum@semihalf.com> +
+Contact: Wojciech Macek <wma@semihalf.com>
+
+The Semihalf team has been working on adding the FreeBSD support for the link:https://www.nxp.com/products/processors-and-microcontrollers/arm-processors/layerscape-processors/layerscape-1028a-applications-processor:LS1028A[NXP LS1028A] SoC, as well as its GPU-less variant (NXP LS1027A).
+
+NXP LS1028A/LS1027A SoC is a dual-core 64-bit ARMv8 Cortex-A72 application processor with high-speed peripherals such as 2 Time-Sensitive Networking-capable (TSN) Ethernet controllers, quad-port TSN-enabled switch, PCIE, SD/MMC, USB3.0 and others.
+
+The original support was extended in a following way:
```
Comment (pauamma) [inline 2021q4/ls1027a.adoc:12]: s/in a/in the/

**Review by pauamma — CHANGES_REQUESTED** [review]
Other that that, LGTM.

**Comment by semihalf-wojtas-marcin** [comment]
> ASLR description seems fine to me although I would expand on the "merged to head" bit - i.e., mention explicitly the current state of being enabled by default on 64-bit architectures [in head].

Done, thanks.

File: 2021q4/ls1027a.adoc (hunk 3 above)
Comment (semihalf-wojtas-marcin) [reply 2021q4/ls1027a.adoc]: Done

File: 2021q4/aslr.adoc (hunk 2 above)
Comment (semihalf-wojtas-marcin) [reply 2021q4/aslr.adoc]: Done. I rephrased the section, please check if it's more clear now.

File: 2021q4/ena.adoc (hunk 1 above)
Comment (semihalf-wojtas-marcin) [reply 2021q4/ena.adoc]: Done

File: 2021q4/ena.adoc [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+Contact: Michal Krawczyk <mk@semihalf.com> +
+Contact: Dawid Gorecki <dgr@semihalf.com> +
+Contact: Marcin Wojtas <mw@semihalf.com>
+
+ENA (Elastic Network Adapter) is the smart NIC available in the virtualized environment of Amazon Web Services (AWS).
+The ENA driver supports multiple transmit and receive queues and can handle up to 100 Gb/s of network traffic, depending on the instance type on which it is used.
+
+Completed since the last update:
+
+* Add IPv6 L4 checksum offload support to the driver
```
[the remark this answers is not in the document]
Comment (semihalf-wojtas-marcin) [reply 2021q4/ena.adoc]: I did s/L4/layer 4/, I hope it's what you requested :)

[the remark this answers is not in the document]
Comment (pauamma) [reply 2021q4/ena.adoc]: > I did s/L4/layer 4/, I hope it's what you requested :)

Yes, "layer 4". My bad.

**Review by pauamma — APPROVED** [review]

**Pull request merged**